pkg/webhooks/machine_webhook.go

```diff
@@ -1517,6 +1521,12 @@ func validateVSphereWorkspace(workspace *machinev1beta1.Workspace, parentPath *f
 		}
 	}
 
+	if config.featureGates.Enabled(featuregate.Feature(apifeatures.FeatureGateVSphereHostVMGroupZonal)) && workspace.VMGroup != "" {
```

Comment:
For this validation, we want the field to not be populated when the gate is not enabled IMO, so, we probably want a check that checks, if the gate is not enabled, and the value is not-empty, return an error.

Then the is the length > 80 check doesn't need to be gated, if I've worked out that logic correctly